src/modules/services/hooks.js

```diff
@@ -58,3 +61,23 @@ export const useWorkflows = () => {
         return { workflowsByType, workflowsLoading };
     }, [serviceWorkflows, workflowsLoading]);
 };
+
+
+export const useMetadataDependentData = () => {
+    const dispatch = useDispatch();
+    const metadata = useBentoService("metadata");
+    useEffect(() => {
+        if (!metadata?.url) return;
+        dispatch(fetchExtraPropertiesSchemaTypes);
+        dispatch(fetchDiscoverySchema);
+    }, [metadata, dispatch]);
+};
+
+export const useGohanDependentData = () => {
```

Comment:
same as with katsu - would be better to connect this hook to where gohan variants overview data is used.